sl_util/sl_util/lang_utils.py

```diff
@@ -0,0 +1,34 @@
+def auto_str(cls):
+    def __str__(self):
+        return '%s(%s)' % (
+            type(self).__name__,
+            ', '.join('%s=%s' % item for item in vars(self).items())
+        )
+
+    cls.__str__ = __str__
+    return cls
+
+
+def auto_repr(cls):
+    def __repr__(self):
+        return '%s(%s)' % (
+            type(self).__name__,
+            ', '.join("%s='%s'" % item for item in vars(self).items())
+        )
+
+    cls.__repr__ = __repr__
+    return cls
+
+
+def auto_eq(cls):
```

Comment:
As discussed with @dantolin-iriusrisk, the `__hash__()` method in this case should be `None`. Python's official documentation says _If a class defines mutable objects and implements an `__eq__()` method, it should not implement `__hash__()`, since the implementation of hashable collections requires that a key’s hash value is immutable (if the object’s hash value changes, it will be in the wrong hash bucket)._